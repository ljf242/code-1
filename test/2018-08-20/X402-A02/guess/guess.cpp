#include<bits/stdc++.h>

using namespace std;

#define REP(i, a, b) for ( LL i = (a), _end_ = (b); i <= _end_; ++ i )
#define mem(a) memset ( (a), 0, sizeof ( a ) )
#define str(a) strlen ( a )
typedef long long LL;

LL a, b, c, m;

inline LL power(LL a, LL b, LL Mod)
{
    LL r = 1;
    while ( b ) 
    {
        if ( b & 1 ) r = r * a % Mod;
        a = a * a % Mod;
        b /= 2;
    }
    return r % Mod;
}

int main()
{
    freopen("guess.in", "r", stdin);
    freopen("guess.out", "w", stdout); 
    int T; scanf("%d", &T);
    while ( T -- )
    {
        scanf("%lld%lld%lld%lld", &m, &a, &b, &c);
        REP(x, 1, m)
            REP(y, 1, m)
            {
                LL sum = (power(x, a, m) + power(y, b, m)) % m;
                REP(z, 1, m)
                    if ( power(z, c, m) == sum ) 
                    {
                        printf("%lld %lld %lld\n", x, y, z);
                        goto yes;
                    }
            }
yes:;
    }
    return 0;
}
