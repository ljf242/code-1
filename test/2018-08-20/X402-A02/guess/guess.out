1 7 2
1 12 13
1 1 4
1 4 7
1 2 3
